tests/test_commands.py

```diff
@@ -4712,6 +4716,34 @@ def test_psync(self, r):
         res = r2.psync(r2.client_id(), 1)
         assert b"FULLRESYNC" in res
 
+    @pytest.mark.onlynoncluster
+    def test_interrupted_command(self, r: redis.Redis):
+        """
+        Regression test for issue #1128:  An Un-handled BaseException
+        will leave the socket with un-read response to a previous
+        command.
+        """
+        print("start")
+
+        def helper():
```

Comment:
```suggestion
        def helper() -> None:
```
IIUC there's no way to get a retval from a thread, though you can pass a mutable object and use that to communicate

Reply:
Also, there is no type hinting in redis-py, not going to start adding that to unit tests :)

Reply:
Yeah, I mostly just meant for you to remove the retval. No need to type-annotate, though I do notice that some stuff is type-annotated.

Reply:
You're right, there is the odd type hinting, but It is not consistently applied.  I guess people add it to help their dev tools with code completion.